src/query/sql/src/planner/metadata.rs

```diff
@@ -55,6 +55,7 @@ pub type MetadataRef = Arc<RwLock<Metadata>>;
 #[derive(Clone, Debug, Default)]
 pub struct Metadata {
     tables: Vec<TableEntry>,
+    view_tables: Vec<TableEntry>,
```

Comment:
Is it redundancy? 

I think `is_view` in `TableEntry` is enough.

Reply:
```
create view default2.v_t20_0012 as select * from default.t20_0012_a;
select * from default2.v_t20_0012;
```

metadata.tables [default.t20_0012_a]
metadata.view_tables [default2.v_t20_0012]

Just verify the view_tables privilege

Reply:
Yeah, so we only need `is_view` in `TableEntry`?

Then
```rust
for table in metadata.tables() {
      if table.is_view() {
             validate_query_privilege(&session, &table).await?;
      }
}
```

Reply:
> Yeah, so we only need `is_view` in `TableEntry`?
> 
> Then
> 
> ```rust
> for table in metadata.tables() {
>       if table.is_view() {
>              validate_query_privilege(&session, &table).await?;
>       }
> }
> ```

metadata.tables() did not contain the TableEntry of the view table.

Reply:
You mean I should push the TableEntry of the ViewTable into to the metadata.tables()?

Reply:
Yes, I didn't think there would be any other problems, you can try～

Reply:
Done, push into metadata.tables, and `source_of_table` means this table is from view table, it should skip permission check.

Reply:
If i push view_table into `metadata.tables`, metadata.tables.len() and metadata.columns.len() will not be equal

Reply:
yeah